src/actions/action.js

```diff
@@ -17,14 +17,17 @@ export default function action(type, host, path, params = {}, transform) {
     });
     const fetchDataWithRetry = delay => fetch(url, path === 'api/metadata' ? { credentials: 'include' } : {})
       .then((response) => {
-        if (!response.ok) {
+        if (!response.ok || !response.status) {
+          const err = new Error();
+          err.fetchError = true;
           dispatch(getError(response.status));
           if (response.status >= 400 && response.status < 500) {
-            const err = new Error('client error');
-            err.clientError = true;
+            err.message = 'fetch failed - client error';
             throw err;
           } else {
-            throw new Error('fetch failed');
+            setTimeout(() => fetchDataWithRetry(delay + 3000), delay);
```

Comment:
if we're retrying why are we also throwing the error?

Reply:
I throw here so the promise chain jumps to the catch handler.

Reply:
I think we should only throw if we exceed some retry threshold.

Reply:
But if I don't throw here, then the next then()  ``` .then(transform || (json => json))```  will throw anyway

Reply:
hmm, ok.  In that case I think we should move the retry back to the catch block.